addons/hr_holidays/static/src/leave_stats/leave_stats.js

```diff
@@ -106,17 +120,39 @@ export class LeaveStatsComponent extends Component {
 
         const dateFrom = date.startOf("year");
         const dateTo = date.endOf("year");
-        this.state.leaves = await this.orm.readGroup(
+
+        const leaves = await this.orm.searchRead(
             "hr.leave",
             [
                 ["employee_id", "=", employee[0]],
                 ["state", "=", "validate"],
                 ["date_from", "<=", dateTo],
                 ["date_to", ">=", dateFrom],
             ],
-            ["holiday_status_id", "number_of_days:sum"],
-            ["holiday_status_id"]
+            ["holiday_status_id", "number_of_days", "number_of_hours", "leave_type_request_unit"]
         );
+
+        const groupedLeaves = leaves.reduce((acc, currentLeave) => {
+            const { holiday_status_id, leave_type_request_unit, number_of_days, number_of_hours } =
+                currentLeave;
+            const currentId = holiday_status_id[0];
+
+            if (!acc[currentId]) {
+                acc[currentId] = {
+                    holiday_status_id,
+                    leave_type_request_unit,
+                    number_of_days,
+                    number_of_hours,
+                };
+            } else {
+                acc[currentId].number_of_days += number_of_days;
+                acc[currentId].number_of_hours += number_of_hours;
+            }
+
+            return acc;
+        }, {});
+
+        this.state.leaves = Object.values(groupedLeaves);
```

Comment:
can't the `readGroup` do that? Is it not working when grouping by `holiday_status_id`?

Reply:
Yes it can and I had implemented a solution with `readGroup` that was something like this.
```js
const leaves = await this.orm.readGroup(
    "hr.leave",
    [
        ["employee_id", "=", employee[0]],
        ["state", "=", "validate"],
        ["date_from", "<=", dateTo],
        ["date_to", ">=", dateFrom],
    ],
    [
        "holiday_status_id",
        "number_of_days",
        "number_of_hours",
        "leave_type_request_unit:array_agg",
    ],
    ["holiday_status_id"]
);
```
Unfortunately it relies on the `array_agg` function and the test framework and/or mock database does not support that function so all the related tests fail. If I remove the function from the domain it doesn't return any `leave_type_request_unit` values, but I will check again and see if I can construct a domain that both solves our problem and works with the tests.

Reply:
Update: I changed the query to use readGroup and converted the failing test to hoot so now it does not fail. The only caveat is that the test does not display days/hours so we can't write a test for that yet but all the existing tests work and pass.